app/src/main/java/com/example/peterchu/watplanner/coursedetail/CourseDetailPresenter.java

```diff
@@ -58,6 +58,7 @@ public void start() {
             }
 
             Course course = dbHandler.getCourse(courseId);
+            if (course == null) { courseDetailFragment.getActivity().finish(); return; }
```

Comment:
Yeah, it's in a weird state, that happened to me twice lol. Sure I'll log error! :)